src/components/Authenticator.js

```diff
@@ -17,14 +18,34 @@ const mockAuth = {
   },
 
   logIn: function() {
-    Cookies.set('user', 'user');
+    signInWithGoogle().then(() => {
+      console.log('User logged in. Downloading their status from server...');
+    });
   },
 
   logOut: function() {
-    Cookies.remove('user');
+    signOutFromGoogle().then(() => {
+      console.log('User logged out.');
+    });
   },
 };
 
-// TODO: Add firebase authentication.
+export const useFirebaseAuthentication = () => {
+  const [authUser, setAuthUser] = useState(null);
+
+  useEffect(() => {
+    return auth.onAuthStateChanged(
+        (
+            newAuthUser => {
+              if (newAuthUser === authUser) {
+
+              } else {
+                setAuthUser(newAuthUser);
+              }
```

Comment:
Maybe this?
```suggestion
              if (newAuthUser !== authUser) {
                setAuthUser(newAuthUser);
              }
```

Reply:
I just realised that this whole if statement can be deleted, because it will be always triggered.

Reply:
I deleted this part, so I'm marking this comment as resolved.